src/Cassandra.IntegrationTests/Core/PoolShortTests.cs

```diff
@@ -176,5 +174,48 @@ public void MarkHostDown_PartialPoolConnection()
                 Assert.IsFalse(h1.IsUp);
             }
         }
+
+        /// <summary>
+        /// Tests that if no host is available at Cluster.Init(), it will initialize next time it is invoked
+        /// </summary>
+        [Test]
+        public void Cluster_Initialization_Recovers_From_NoHostAvailableException()
+        {
+            var testCluster = TestClusterManager.CreateNew();
+            testCluster.StopForce(1);
+            var cluster = Cluster.Builder()
+                                 .AddContactPoint(testCluster.InitialContactPoint)
+                                 .Build();
+            //initially it will throw as there is no node reachable
+            Assert.Throws<NoHostAvailableException>(() => cluster.Connect());
+
+            // wait for the node to be up
+            testCluster.Start(1);
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
```

Comment:
Could use Stopwatch.StartNew() to create instance and starts. Also we need to stopped after the succeeded connection and define a reasonable time to assert around it.